appIconIndicators.js

```diff
@@ -794,7 +794,19 @@ const UnityIndicator = new Lang.Class({
             return;
         }
 
-        this._progressOverlayArea = new St.DrawingArea({x_expand: true, y_expand: true});
+        this._progressOverlayArea = new St.DrawingArea({name: 'dashtodockContainer', x_expand: true, y_expand: true});
+
+        // Get progress-bar background and border color from CSS
+        this._progressOverlayArea.add_style_class_name('progress-bar');
+        Main.uiGroup.add_child(this._progressOverlayArea);
+
```

Comment:
What I had in mind, was to apply permanently the ` progress-bar` class directly to the `stDrawinArea`, but then we cannot use the standard css properties as they would be rendered on the widget itself. However, we could use custom style defintion (someting lik `-progress-bar-background', or some better naming...), and then read via `them_node.get_color('-progress-bar-background')`. This way we avoid all this machinery of adding/removing elements, and we make clear that it a special styling.

Reply:
That's better for sure. I didn't think about custom style definition